dask_cuda/explicit_comms/dataframe/shuffle.py

```diff
@@ -23,9 +22,72 @@
 T = TypeVar("T")
 
 
+class Proxify(Protocol):
+    """Proxify type hint"""
+
+    def __call__(self, obj: T) -> T:
+        ...
```

Comment:
```
from typing import Callable
...
Proxify = Callable[[T], T]
```
?

I see this is what you used to have, so there's presumably some reason it's no longer sufficient?

Reply:
It was mypy that compiled at some point but it seems to accept `Proxify = Callable[[T], T]` now. I must have changed something :)

Anyways, I think we should begin using mypy in CI: https://github.com/rapidsai/dask-cuda/issues/1077